### DISCUSSION TITLE
bzlws_copy not copying the files during build

### DISCUSSION THREAD

**Discussion by dDipam** [zaucy/bzlws#12] [Q&A] [answered]
When I tried bzlws_copy example from the repository it did not copy the "example.txt" file during build of that target but instead created an executable, which on execution copies "example.txt" file to the given out path. Is this the expected outcome or building the target should have copied the file?

**Comment by zaucy** [comment, marked as answer]
That is the expected outcome. Instead of building you're expected to _run_ `bzlws_*` rules with `bazel run`. `bzlws` is for copying files _into_ your workspace which is not possible at build time. I will put this disclaimer on the README. Tracking here https://github.com/zaucy/bzlws/issues/13

If you're looking for a way to copy files at build time I might suggest checking out [bazel skylib](https://github.com/bazelbuild/bazel-skylib) specifically the `copy_file` rule.

**Reply by dDipam** [reply]
Thank You!
